addons/payment_asiapay/controllers/main.py

```diff
@@ -0,0 +1,34 @@
+import logging
+import pprint
+
+from odoo import _, http
+from odoo.exceptions import ValidationError
+from odoo.http import request
+
+_logger = logging.getLogger(__name__)
+
+class AsiaPayController(http.Controller):
+    _return_url = "/payment/asiapay/return"
+    _notify_url = "/payment/asiapay/notify"
+
+    @http.route(_return_url, type="http", auth="public", methods=['GET'])
+    def asiapay_return(self, **data):
+        return request.redirect('/payment/status')
+
+    @http.route(_notify_url, type="http", auth="public", methods=['POST'], csrf=False)
+    def asiapay_notify(self, **post):
+        _logger.info("received AsiaPay notification data:\n%s", pprint.pformat(post))
+        self._asiapay_validate_notification(**post)
+        request.env['payment.transaction'].sudo()._handle_feedback_data('asiapay', post)
```

Comment:
This should typically be surrounded by a try/except (ValidationError) clause to discard the notification and reply with an HTTP 200 response if something goes wrong while processing the feedback data. This prevents the provider from shutting down the webhook because of a single transaction processing repeatedly failing in Odoo.